### PR TITLE
creating rules around taskRuns

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: policies/taskrun_test.rego [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+test_allow_task_kind_denied {
+	allow_task_kind with data.kind as "TaskRun"
+}
+
+test_allow_task_name {
+	not allow_task_name with data.metadata as {"name": "mybadtask"}
+}
+
+test_sllow_task_name_denied {
```
Comment (simonbaird) [inline policies/taskrun_test.rego:15]: Typo "sllow"

Comment (simonbaird) [reply policies/taskrun_test.rego:15]: Also maybe line 11 should be "..._denied" this one should be "..._allowed" ..?

Comment (joejstuart) [reply policies/taskrun_test.rego]: Yes, I had those backwards. Fixed!

**Comment by simonbaird** [comment]
Code looks good, but I think it needs to be reworked a little so the keys are based on attestation data format rather than the cluster data format.

**Comment by simonbaird** [comment]
Need to run `make fmt`

File: policies/taskrun.rego [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package hacbs.contract.taskrun
+
+validAnnotations := ["Red Hat", null]
+validRegistries := ["docker.io", "registry.access.redhat.com"]
```
Comment (simonbaird) [inline policies/taskrun.rego:4]: An idea: Could put these lists in policies/lib/config.rego .

There are pros and cons, but something to consider.

Comment (joejstuart) [reply policies/taskrun.rego:4]: I will look at that. I need to rework the tests also.

**Comment by joejstuart** [comment]
Closing this in favor of https://github.com/hacbs-contract/ec-policies/pull/12. I will add the annotation check once that is merged.

**Comment by simonbaird** [comment]
No harm in making a PR, but fwiw I think this is good enough to rebase and reopen.

**Pull request closed without merging**